test/unit/lodash.spec.js

```diff
@@ -0,0 +1,1080 @@
+            });
+            it('should square even numbers otherwise add +1 to the odd numbers in the array', function () {
+                expect(_.map.apply(_, datasets.map2())).to.eql([2, 4, 4, 16, 6, 36]);
+            });
+        });
+    });
+
+    describe('partition', function () {
+        var users = [
+            {name: 'Andrinio', age: 36, active: false},
+            {name: 'Harry', age: 40, active: true},
+            {name: 'Kodi', age: 1, active: false},
+            {name: 'Francheska', age: 61, active: true}
+        ];
+
+        describe('partition function', function () {
+            var params = datasets.partition(users);
+            var spy = sinon.spy(params[1]);
+
+            it('should match types of passing elements', function () {
+                var element = params[1] instanceof Function;
+
+                expect(element).to.eql(true);
+                expect(params).to.have.length(2);
+            });
+            it('should pass mock test', function () {
+                var testObj = [
+                    {qq: '123', ee: 10, active: false},
+                    {qq: '456', ee: 40, active: true}
+                ];
+                var checkFunc = _.partition.apply(_, [testObj, spy]);
+
+                expect(spy).to.have.been.callCount(2);
+                expect(checkFunc).to.eql([
+                    [{qq: '456', ee: 40, active: true}],
+                    [{qq: '123', ee: 10, active: false}]
+                ]);
+            });
+            it('should split persons to two tables using active property', function () {
+                expect(_.partition.apply(_, datasets.partition(users))).to.eql([
+                    [{name: 'Harry', age: 40, active: true}, {name: 'Francheska', age: 61, active: true}],
+                    [{name: 'Andrinio', age: 36, active: false}, {name: 'Kodi', age: 1, active: false}]
+                ]);
+            });
+        });
+        describe('partition2 function', function () {
+            var params = datasets.partition2(users);
+
+            it('should match types of passing elements', function () {
+                var element = typeof params[1];
+                expect(element).to.eql('string');
+                expect(element).to.have.length.above(1);
+                expect(params).to.have.length(2);
+            });
+            it('should split persons only to truthy table when predicate is true for all', function () {
+                expect(_.partition.apply(_, datasets.partition2(users))).to.eql(
+                    [
+                        [{name: 'Andrinio', age: 36, active: false},
+                            {name: 'Harry', age: 40, active: true},
+                            {name: 'Kodi', age: 1, active: false},
+                            {name: 'Francheska', age: 61, active: true}], []
+                    ]);
+            });
+        });
+        describe('partition3 function', function () {
```

Comment:
With empty array and function only last test not passed, making this set impossible to solving